source/scale/high-availability-cluster-based-deployment.rst

```diff
@@ -8,462 +8,414 @@ High availability cluster-based deployment
 
-  Specifying configuration setting values using Mattermost environment variables ensure that they always take precedent over any other configuration settings.
+Transparent failover: Transparent failover setups are beyond the scope of this document. 
 
-For an AWS High Availability RDS cluster deployment, point the :ref:`datasource <configure/environment-configuration-settings:data source>` configuration setting to the write/read endpoint at the **cluster** level to benefit from the AWS failover handling. AWS takes care of promoting different database nodes to be the writer node. Mattermost doesn't need to manage this. 
+Automatic failover: use database technologies such as PostgreSQL Clustering or Amazon Aurora.
 
-Use the :ref:`read replica <configure/environment-configuration-settings:read replicas>` feature to scale the database. The Mattermost server can be set up to use one master database and one or more read replica databases. 
+ 
+PostgreSQL optimization (Optional)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. note::
-  
-  For an AWS High Availability RDS cluster deployment, don't hard-code the IP addresses. Point this configuration setting to the write/read endpoint at the **cluster** level. This will benefit from the AWS failover handling where AWS takes care of promoting different database nodes to be the writer node. Mattermost doesn't need to manage this. 
+For improved performance, apply the following recommendations:
 
-On large deployments, also consider using the :ref:`search replicas <configure/environment-configuration-settings:search replicas>` feature to isolate search queries onto one or more search replicas. A search replica is similar to a read replica, but is used only for handling search queries.
+Primary Node:
 
-.. note::
+- ``max_connections = 1024``
+- ``random_page_cost = 1.1``
+- ``effective_cache_size = 21GB``
+- ``shared_buffers = 21GB``
+- ``tcp_keepalives_idle = 5``
+- ``tcp_keepalives_interval = 1``
+- ``tcp_keepalives_count = 5``
+- ``maintenance_work_mem = 512MB``
+- ``autovacuum_max_workers = 4``
+- ``autovacuum_vacuum_cost_limit = 500``
+- ``max_worker_processes = 12``
+- ``max_parallel_workers_per_gather = 4``
+- ``max_parallel_workers = 12``
+- ``max_parallel_maintenance_workers = 4``
 
-  For an AWS High Availability RDS cluster deployment, don't hard-code the IP addresses. Point this configuration setting directly to the underlying read-only node endpoints within the RDS cluster. We recommend circumventing the failover/load balancing that AWS/RDS takes care of (except for the write traffic). Mattermost has its own method of balancing the read-only connections, and can also balance those queries to the DataSource/write+read connection should those nodes fail. 
+Replica Node (in addition to the primary node settings above):
 
-Mattermost distributes queries as follows:
+- ``work_mem = 16MB``
+- ``hot_standby = on``
+- ``hot_standby_feedback = on``
```

Comment:
We've lost all the context that was in the comments of the code block in the old document. There are also missing items, like `work_mem` for the master node.